File: python/2020/day01.py
values = []

with open('./day01.txt') as file:
    for line in file:
        values.append(int(line))

for i in range(0, len(values) - 1):
    for j in range(i, len(values)):
        if values[i]+values[j] == 2020:
            print(values[i]*values[j])

for i in range(0, len(values) - 2):
    for j in range(i, len(values) - 1):
        for k in range(j, len(values)):
            if values[i]+values[j]+values[k] == 2020:
                print(values[i]*values[j]*values[k])

File: java/src/main/java/com/adrean/aoc/year2020/Day01.java
package com.adrean.aoc.year2020;

import com.adrean.aoc.io.InputReader;

import java.io.BufferedReader;
import java.util.stream.Stream;

public class Day01 {
    public static void main(String[] args) {
        BufferedReader reader = InputReader.getReader("2020/day01.txt");

        // Part 1
        Integer[] values = reader.lines().map(Integer::parseInt).toArray(Integer[]::new);
        for (int i=0; i < values.length - 1; i++ ){
            for (int j=i+1; j < values.length; j++) {
                if (values[i] + values[j] == 2020) {
                    System.out.println(values[i] * values[j]);
                }
            }
        }

        // Part 2
        for (int i=0; i < values.length - 2; i++ ){
            for (int j=i+1; j < values.length - 1; j++) {
                for (int k=j; k < values.length; k++){
                    if (values[i] + values[j] + values[k] == 2020) {
                        System.out.println(values[i] * values[j] * values[k]);
                    }
                }
            }
        }

    }
}